amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -409,13 +410,17 @@ type ColonyAction @model {
   type: ColonyActionType!
   blockNumber: Int!
   createdAt: AWSDateTime!
-  initiatorAddress: ID
-  initiator: User @hasOne(fields: ["initiatorAddress"])
+  # Action type specific fields which might be null
+  initiatorAddress: ID # Amplify will automatically populate one of the following fields with related model if there is an initiator
+  initiatorUser: User @hasOne(fields: ["initiatorAddress"])
+  initiatorExtension: ColonyExtension @hasOne(fields: ["initiatorAddress"])
+  initiatorColony: Colony @hasOne(fields: ["initiatorAddress"])
```

Comment:
Unfortunately union connections are not supported by Amplify, so we can't do it cleaner like:
```gql
initiator: User | ColonyExtension | Colony @hasOne(fields: ["initiatorAddress"])
```